tutorials/xarray_tutorial.py

```diff
@@ -6,295 +6,344 @@
 
 #########################################################################
-# Also, a limited number of calculations directly support xarray DataArrays or Datasets (they
-# can accept *and* return xarray objects). Right now, this includes
-#
-# - Derivative functions
-#     - ``first_derivative``
-#     - ``second_derivative``
-#     - ``gradient``
-#     - ``laplacian``
-# - Cross-section functions
-#     - ``cross_section_components``
-#     - ``normal_component``
-#     - ``tangential_component``
-#     - ``absolute_momentum``
-# - Smoothing functions
-#     - ``smooth_gaussian``
-#     - ``smooth_n_point``
-#     - ``smooth_window``
-#     - ``smooth_rectangular``
-#     - ``smooth_circular``
-#
-# More details can be found by looking at the documentation for the specific function of
-# interest.
+# Accessing just the underlying units:
+height_units = heights.metpy.units
+height_units
 
 #########################################################################
-# There is also the special case of the helper function, ``grid_deltas_from_dataarray``, which
-# takes a ``DataArray`` input, but returns unit arrays for use in other calculations. We could
-# rewrite the above geostrophic wind example using this helper function as follows:
+# Calculations
+# ------------
+#
+# MetPy's xarray integration extends to its calcuation suite as well. Most grid-capable
+# calculations (such as thermodynamics, kinematics, and smoothers) fully support xarray
+# ``DataArray``\s by accepting them as inputs, returning them as outputs, and automatically
+# using the attached coordinate data/metadata to determine grid arguments
+
+heights = data_parsed.metpy.parse_cf('Geopotential_height_isobaric').metpy.sel(
+    time='2017-09-05 18:00',
+    vertical=50000.
+)
+u_g, v_g = mpcalc.geostrophic_wind(heights)
+u_g
 
-heights = data['height'].metpy.loc[{'time': time[0], 'vertical': 500. * units.hPa}]
-lat, lon = xr.broadcast(y, x)
-dx, dy = grid_deltas_from_dataarray(heights)
-u_geo, v_geo = mpcalc.geostrophic_wind(heights, dx, dy, lat)
-print(u_geo)
-print(v_geo)
+#########################################################################
+# For profile-based calculations (and most remaining calculations in the ``metpy.calc``
+# module), xarray ``DataArray``\s are accepted as inputs, but the outputs remain Pint
+# Quantities (typically scalars)
+
+data_at_point = data.metpy.sel(
+    time1='2017-09-05 12:00',
+    latitude=40 * units.degrees_north,
+    longitude=260 * units.degrees_east
+)
+dewpoint = mpcalc.dewpoint_from_relative_humidity(
+    data_at_point['Temperature_isobaric'],
+    data_at_point['Relative_humidity_isobaric']
+)
+cape, cin = mpcalc.surface_based_cape_cin(
+    data_at_point['isobaric3'],
+    data_at_point['Temperature_isobaric'],
+    dewpoint
+)
+cape
 
 #########################################################################
-# Plotting
-# --------
+# Remaining portions of MetPy's calculations (such as its interpolation module) do not fully
```

Comment:
Isn't `interpolation` really it?

Reply:
Not quite, `mpcalc.resample_nn_1d` and `mpcalc.nearest_intersection_idx` don't have the decorator at all, and I'm not sure that every function that accepts DataArrays as input but only as downcasted to Pint Quanities falls under the previously discussed profile-based calculations, so this was a loose catch-all statement. But, if we need something more precise, that's fine too!

Reply:
Maybe "a few remaining portions...(mainly the interpolation module and a few other functions)"?